### PR TITLE
Update client to use Poison decoding capabilities

### PR DESCRIPTION
**Pull request by jacegu**
This is a follow up of #49

### CODE REVIEW

**Comment by jacegu** [comment]
Worth noting over what was already stated on #49 is the need to specify decoding of pagination information. Although there is ways to prevent this I think it makes it obvious which endpoints are paginated and which aren't, which is a nice addition.

File: lib/dnsimple/utils.ex [hunk 1]
```diff
@@ -1,67 +1,5 @@
...
-  defp value(key, value, nested) do
-    case Keyword.get(nested, key) do
-      nil         -> value
-      nested_kind -> attrs_to_struct(value, nested_kind)
-    end
-  end
-
-
   # http://michal.muskala.eu/2015/07/30/unix-timestamps-in-elixir.html
   defmodule DateTime do
```
Comment (jacegu) [inline lib/dnsimple/utils.ex]: @weppos at this point the only thing remaining in the `Dnsimple.Utils` module is this `DateTime` submodule. I've checked and it's not used anywhere. Can it be removed?

Comment (jacegu) [inline lib/dnsimple/utils.ex]: @weppos I've finally removed it. If we ever need to add it back we can always dig it back from git.

**Comment by jacegu** [comment]
Last opportunity to review this one: merging in 5 😇

**Comment by jodosha** [comment]
@jacegu Better to have a Community maintained lib for JSON parsing, rather than a custom one. :shipit:

**Pull request merged**